app/uk/gov/hmrc/helptosavefrontend/controllers/HelpAndContactController.scala

```diff
@@ -0,0 +1,95 @@
+                                          http:                  WSHttp)(implicit ec: ExecutionContext)
+  extends HelpToSaveAuth(frontendAuthConnector, metrics) with HelpToSaveFrontendController with I18nSupport with Logging {
+
+  val contactFormServiceIdentifier: String = "HTS"
+
+  val TICKET_ID: String = "ticketId"
+
+  val submitUrl: String = routes.HelpAndContactController.submitContactHmrcForm().url
+
+  val contactHmrcFormPartialUrl: String = s"$contactUrl/contact-hmrc/form?service=$contactFormServiceIdentifier" +
+    s"&submitUrl=${urlEncode(submitUrl)}"
+
+  val contactHmrcSubmitPartialUrl: String = s"$contactUrl/contact-hmrc/form?resubmitUrl=${urlEncode(submitUrl)}"
+
+  private def contactHmrcThankYouPartialUrl(ticketId: String) =
+    s"$contactUrl/contact-hmrc/form/confirmation?ticketId=${urlEncode(ticketId)}"
+
+  def getHelpAndContactPage: Action[AnyContent] = authorisedForHts { implicit request ⇒ implicit htsContext ⇒
+    Ok(views.html.contact_hmrc(contactHmrcFormPartialUrl, None, formProvider))
+  }(redirectOnLoginURL = routes.HelpAndContactController.getHelpAndContactPage().url)
+
+  def submitContactHmrcForm: Action[AnyContent] = authorisedForHts { implicit request ⇒ implicit htsContext ⇒
+    submitContactHmrc(contactHmrcSubmitPartialUrl,
+                      routes.HelpAndContactController.contactHmrcThankYou(),
+      body ⇒ views.html.contact_hmrc(contactHmrcFormPartialUrl, Some(body), formProvider))
+  }(redirectOnLoginURL = routes.HelpAndContactController.getHelpAndContactPage().url)
+
+  def contactHmrcThankYou: Action[AnyContent] = authorisedForHts { implicit request ⇒ implicit htsContext ⇒
+    val ticketId = request.session.get(TICKET_ID).getOrElse("N/A")
+    Ok(views.html.contact_hmrc_thankyou(contactHmrcThankYouPartialUrl(ticketId), formProvider))
+  }(redirectOnLoginURL = routes.HelpAndContactController.getHelpAndContactPage().url)
+
+  private def submitContactHmrc(formUrl:                         String,
+                                successRedirect:                 Call,
+                                failedValidationResponseContent: (Html) ⇒ HtmlFormat.Appendable)(
+      implicit
+      request: Request[AnyContent]): Future[Result] = {
+    request.body.asFormUrlEncoded.fold[Future[Result]] {
+      logger.warn("Trying to submit an empty contact form")
+      internalServerError()
+    }{ formData ⇒
+      http.postForm(formUrl, formData.mapValues(_.toList))
+        .map {
+          resp ⇒
+            resp.status match {
+              case 200 ⇒ Redirect(successRedirect).withSession(request.session + (TICKET_ID -> resp.body))
+              case 400 ⇒ BadRequest(failedValidationResponseContent(Html(resp.body)))
+              case 500 ⇒ InternalServerError(Html(resp.body))
+              case status ⇒
+                logger.warn(s"Unexpected status code from contact HMRC form: $status")
+                Status(status)(Html(resp.body))
+            }
+        }
```

Comment:
just in case, it might be worth writing a `.recover` block for this future in case it fails